### PR TITLE
[DOC] added post-processing notes in tutorial

### PR DESCRIPTION
**Pull request by nshahpazov**
### Reference Issues/PRs
- https://github.com/sktime/sktime/issues/2608
### Description of PR

- [x] Checked and confirmed that dunder pipeline methods are described in the tutorial
- [x] Added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/.all-contributorsrc).
- [x] Added description in the tutorial on how TransformedTargetForecaster can also end with a post-processing Transformer, in this case, I've used a simple predictions rounding

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
That´s nice, thanks!

May I ask for one thing, to keep the repository size small: can you make sure only the parts are changed where you add content?

I think the current huge diff comes from you rerendering plots.

Worst case, this might require going into the ipynb file with an editor, but should also work if you take the original file and copy-paste the change.

**Comment by nshahpazov** [comment]
@fkiraly I've addressed the proposed changes. The actual diff is visible now.

File: examples/01_forecasting.ipynb [hunk 1]
```diff
@@ -5376,7 +5376,8 @@
    "source": [
```
Comment (aiwalter) [inline examples/01_forecasting.ipynb:5376]: <p>we actually dont have a scitype "post-processing transformer", so maybe lets frame that a little bit different?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sktime/sktime/pull/3878/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='203'/>

**Comment by aiwalter** [comment]
also please make sure you run pre-commit bcz linting is failing

File: examples/01_forecasting.ipynb [hunk 2]
```diff
@@ -5425,7 +5426,7 @@
...
+    "The `TransformedTargetForecaster` is constructed with a list of steps, each a pair of name and estimator. The last estimator should be of either, a forecaster scitype or a post-processing transformer, the other estimators should be series-to-series transformers which possess both a `transform` and `inverse_transform` method. The resulting estimator is of forecaster scitype and has all interface defining methods. In `fit`, all transformers apply `fit_transforms` to the data, then the forecaster's `fit`; in `predict`, first the forecaster's `predict` is applied, then the transformers' `inverse_transform` in reverse order."
```
[the remark this answers is not in the document]
Comment (fkiraly) [reply examples/01_forecasting.ipynb:5429]: bit complex, so I would suggest to split the explanation:

* case 1 - only pre-processing transformers
* case 2 - only post-processing transformers
* case 3 - putting it together, and spelling out the sequence

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks! Great idea to add this content, will be useful to our users!

Change requests:

* linting
* I would split the content and introduce it step by step, or it may be confusing if it comes all at once, see suggestion above

File: examples/01_forecasting.ipynb (hunk 1 above)
Comment (nshahpazov) [reply examples/01_forecasting.ipynb]: Thanks!
I've the post-processing from this paragraph and moved it further below, following the other comments.

**Comment by nshahpazov** [comment]
@fkiraly I've rephrased the paragraphs into three sections like you suggested. Linting is working now. Let me know if you find it ok.

**Review by fkiraly — APPROVED** [review]
All good! Just made some minor changes in formatting.

**Review by fkiraly — APPROVED** [review]

**Review by lmmentel — APPROVED** [review]

**Pull request merged**